assets/src/components/vehicleIcon.tsx

```diff
@@ -46,35 +47,48 @@ const Y_CENTER_TO_POINT = 20
 const Y_CENTER_TO_BASE = 20
 const ALERT_ICON_RADIUS = 27
 
-export const VehicleIcon = (props: Props): ReactElement<HTMLElement> => {
-  if (
-    props.status === "ghost" &&
-    (props.orientation === Orientation.Left ||
-      props.orientation === Orientation.Right)
-  ) {
-    props = {
-      ...props,
-      // ghosts can't be drawn sideways
-      orientation:
-        props.status === "ghost" ? Orientation.Up : props.orientation,
+export const VehicleIcon = React.memo(
+  (props: Props): ReactElement<HTMLElement> => {
+    if (
+      props.status === "ghost" &&
+      (props.orientation === Orientation.Left ||
+        props.orientation === Orientation.Right)
+    ) {
+      props = {
+        ...props,
+        // ghosts can't be drawn sideways
+        orientation:
+          props.status === "ghost" ? Orientation.Up : props.orientation,
+      }
     }
-  }
 
-  const { left, top, width, height } = viewBox(props)
-  return (
-    <svg
-      style={{ width, height }}
-      viewBox={`${left} ${top} ${width} ${height}`}
-    >
-      <VehicleIconSvgNode {...props} />
-    </svg>
-  )
-}
+    const { left, top, width, height } = viewBox(props)
+    return (
+      <svg
+        style={{ width, height }}
+        viewBox={`${left} ${top} ${width} ${height}`}
+      >
+        <VehicleIconSvgNode {...props} />
+      </svg>
+    )
+  }
+)
 
 export const VehicleTooltip = ({
   vehicleOrGhost,
   children,
 }: TooltipProps): ReactElement<HTMLElement> => {
```

Comment:
Just curious, what's the reason for not memoizing this one directly? Does it have something to do with `children` being hard to compare for memoization purposes?

Reply:
The logic below that calculates `label`, etc. has to live somewhere. If not here, then we'd have to repeat it in the multiple places that we call `VehicleTooltip`--not very DRY. So we're kind of stuck with `vehicleOrGhost` here, unless we want to do a much bigger refactoring, and as a rule of thumb I've assumed that `Vehicle`s and `Ghost`s will change too often to be worth memoizing.